descartes_light/solvers/include/descartes_light/solvers/bgl/impl/event_visitors.hpp

```diff
@@ -32,6 +32,82 @@ struct early_terminator : public boost::base_visitor<early_terminator<FloatType>
   const long last_rung_idx_;
 };
 
+/**
+ * @brief Event visitor that adds all edges to each node discovered on a depth first search
+ * @details Throws the vertex descriptor that is the termination of the path once a vertex in the last rung of
+ * the graph is encountered
+ */
+template <typename FloatType>
+struct add_all_dfs : public boost::base_visitor<add_all_dfs<FloatType>>
+{
+  /** @brief Event filter typedef defining the events for which this visitor can be used */
+  typedef boost::on_discover_vertex event_filter;
+
+  add_all_dfs(std::vector<typename EdgeEvaluator<FloatType>::ConstPtr>& edge_eval,
+              std::vector<std::vector<VertexDesc<FloatType>>>& ladder_rungs)
+    : eval_(edge_eval), ladder_rungs_(ladder_rungs)
+  {
+  }
+
+  void operator()(VertexDesc<FloatType> u, const BGLGraph<FloatType>& g)
+  {
+    auto out_deg = boost::out_degree(u, g);
+    // return if the vertex has any out edges
+    if (out_deg == 0)
```

Comment:
Remove the aspect of this visitor that throws the vertex descriptor if the vertex is in the last rung. If a user wants that capability, they should simply chain together this event visitor and the `early_terminator` visitor